Fix for spinboxes that break when focused in Tk 8.6.9
See https://core.tcl.tk/tk/tktview/273b6a4996
Patch from chw's comment dated 2018-11-19 06:16:24

Index: macosx/tkMacOSXEntry.c
==================================================================
--- macosx/tkMacOSXEntry.c
+++ macosx/tkMacOSXEntry.c
@@ -147,10 +147,13 @@
     bounds.origin.x = macDraw->xOff + MAC_OSX_FOCUS_WIDTH;
     bounds.origin.y = macDraw->yOff + MAC_OSX_FOCUS_WIDTH;
     bounds.size.width = Tk_Width(tkwin) - 2*MAC_OSX_FOCUS_WIDTH;
     bounds.size.height = Tk_Height(tkwin) - 2*MAC_OSX_FOCUS_WIDTH;
     if (!TkMacOSXSetupDrawingContext(d, NULL, 1, &dc)) {
+	if (isSpinbox) {
+	    Tk_Width(tkwin) = oldWidth;
+	}
 	return 0;
     }
     ChkErr(HIThemeDrawFrame, &bounds, &info, dc.context, HIOrientation);
     TkMacOSXRestoreDrawingContext(&dc);
     if (isSpinbox) {
